Source/Scene/Material.js

```diff
@@ -853,13 +854,29 @@ function createTexture2DUpdateFunction(uniformId) {
       return;
     }
 
+    if (uniformChanged && defined(texture)) {
+      if (
+        texture !== this.defaultTexture &&
+        texture !== context.defaultTexture
+      ) {
+        texture.destroy();
+      }
+
+      if (uniformValueIsDefaultImage) {
+        texture = undefined;
+      }
+    }
+
     if (!defined(texture)) {
       material._texturePaths[uniformId] = undefined;
       if (!defined(material._defaultTexture)) {
         material._defaultTexture = context.defaultTexture;
       }
-      texture = material._textures[uniformId] = material._defaultTexture;
-
+      if (uniformValueIsDefaultImage) {
+        texture = material._textures[uniformId] = context.defaultTexture;
+      } else {
+        texture = material._textures[uniformId] = material._defaultTexture;
+      }
```

Comment:
Since `material._defaultTexture` is set to `context.defaultTexture` above I don't think this if/else is needed. The previous code can be used.

```
texture = material._textures[uniformId] = material._defaultTexture;
```